lib/ipmi_hpeoem.c

```diff
@@ -0,0 +1,428 @@
+            return -1;
+        }
+        if (str2uchar(argv[current_arg], &channel) != 0)
+        {
+            lprintf(LOG_ERR, "Invalid channel: %s", argv[current_arg]);
+            return -1;
+        }
+
+        current_arg++;
+        nic_selection = ipmi_hpeoem_parse_nic_selection_mode(argv);
+        if (nic_selection == INVALID)
+        {
+            lprintf(LOG_ERR, "Invalid NIC selection mode");
+            ipmi_hpeoem_lan_usage();
+            return -1;
+        }
+
+        ipmi_hpeoem_lan_set_nic_selection(intf, channel, nic_selection);
+        return 0;
+    }
+    else if (!strcmp(argv[current_arg], "get"))
+    {
+        current_arg++;
+        if (!argv[current_arg])
+        {
+            ipmi_hpeoem_lan_usage();
+            return -1;
+        }
+        if (str2uchar(argv[current_arg], &channel) != 0)
+        {
+            lprintf(LOG_ERR, "Invalid channel: %s", argv[current_arg]);
+            return -1;
+        }
+        ipmi_hpeoem_lan_get_nic_selection(intf, channel);
+    }
+    else
+    {
+        ipmi_hpeoem_lan_usage();
+        return -1;
+    }
+    return rc;
+}
+
+int ipmi_hpeoem_main(struct ipmi_intf *intf, int argc, char **argv)
+{
+    int rc = 0;
+    current_arg = 0;
+    if (!argc || !strcmp(argv[0], "help"))
+    {
+        usage();
+        return 0;
+    }
+    if (!strcmp(argv[current_arg], "lan"))
+    {
+        /* lan address */
+        rc = ipmi_hpeoem_lan_main(intf, argc, argv);
+    }
+    else
+    {
+        usage();
+        return -1;
+    }
+    return rc;
+}
+
+static void
+usage(void)
+{
+    lprintf(LOG_NOTICE,
+            "");
+    lprintf(LOG_NOTICE,
+            "usage: hpeoem <command> [option...]");
+    lprintf(LOG_NOTICE,
+            "");
+    lprintf(LOG_NOTICE,
+            "commands:");
+    lprintf(LOG_NOTICE,
+            "    lan");
+    lprintf(LOG_NOTICE,
+            "");
+    lprintf(LOG_NOTICE,
+            "For help on individual commands type:");
+    lprintf(LOG_NOTICE,
+            "hpeoem <command> help");
+}
+
+static void
+ipmi_hpeoem_lan_usage(void)
+{
+    lprintf(LOG_NOTICE,
+            "");
+    lprintf(LOG_NOTICE,
+            "   lan set <channel> <mode>");
+    lprintf(LOG_NOTICE,
+            "      sets the NIC Selection Mode :");
+    lprintf(LOG_NOTICE,
+            "            dedicated, shared with lom1, shared with lom2, shared with alom1,");
+    lprintf(LOG_NOTICE,
+            "            shared with alom2");
+    lprintf(LOG_NOTICE,
+            "   lan get <channel>");
+    lprintf(LOG_NOTICE,
+            "      returns the current active NIC.");
+    lprintf(LOG_NOTICE,
+            "");
+}
+
+static int
+ipmi_hpeoem_lan_get_nic_selection(struct ipmi_intf *intf, int chan)
+{
+    struct ipmi_rs *rsp;
+    struct ipmi_rq req;
+    uint8_t input_length = 4;
+    uint8_t msg_data[4];
+    uint8_t nic_selection = -1;
+    uint8_t shared_nic_selection = -1;
+    uint8_t shared_nic_port = -1;
```

Comment:
Don't all those `-1` values stand for `INVALID` from line 71 above? Please use that macro/enum instead. If you go there with enum (preferred), then please also update the variable types here.